ironfish/src/assets/assetsVerifier.test.ts

```diff
@@ -0,0 +1,258 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+import nock from 'nock'
+import { AssetsVerifier } from './assetsVerifier'
+
+/* eslint-disable jest/no-standalone-expect */
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+describe('AssetsVerifier', () => {
+  jest.useFakeTimers()
+
+  const waitForRefreshToFinish = async (refreshSpy: jest.SpyInstance) => {
+    for (const result of refreshSpy.mock.results) {
+      await result.value
+    }
+  }
+
+  beforeEach(() => {
+    nock.cleanAll()
+    jest.clearAllTimers()
+  })
+
+  afterEach(() => {
+    expect(nock.pendingMocks()).toHaveLength(0)
+  })
+
+  it('does not refresh when not started', () => {
+    const assetsVerifier = new AssetsVerifier()
+    const refresh = jest.spyOn(assetsVerifier as any, 'refresh')
+
+    jest.runOnlyPendingTimers()
+    expect(refresh).toHaveBeenCalledTimes(0)
+  })
+
+  it('periodically refreshes once started', async () => {
+    nock('https://test')
+      .get('/assets/verified')
+      .reply(200, {
+        data: [{ identifier: '0123' }],
+      })
+      .get('/assets/verified')
+      .reply(200, {
+        data: [{ identifier: '4567' }],
+      })
+      .get('/assets/verified')
+      .reply(200, {
+        data: [{ identifier: '89ab' }],
+      })
+
+    const assetsVerifier = new AssetsVerifier({
+      apiUrl: 'https://test/assets/verified',
+    })
+    const refresh = jest.spyOn(assetsVerifier as any, 'refresh')
+
+    expect(assetsVerifier.verify('0123')).toStrictEqual({ status: 'unknown' })
+    expect(assetsVerifier.verify('4567')).toStrictEqual({ status: 'unknown' })
+    expect(assetsVerifier.verify('89ab')).toStrictEqual({ status: 'unknown' })
+
+    assetsVerifier.start()
+    expect(refresh).toHaveBeenCalledTimes(1)
+    await waitForRefreshToFinish(refresh)
+
+    expect(assetsVerifier.verify('0123')).toStrictEqual({ status: 'verified' })
+    expect(assetsVerifier.verify('4567')).toStrictEqual({ status: 'unverified' })
+    expect(assetsVerifier.verify('89ab')).toStrictEqual({ status: 'unverified' })
+
+    jest.runOnlyPendingTimers()
+    expect(refresh).toHaveBeenCalledTimes(2)
+    await waitForRefreshToFinish(refresh)
```

Comment:
Really cool pattern! I don't think we have good examples of using mock timers in our codebase.